clock-bound-d/src/server.rs

```diff
@@ -78,15 +79,9 @@ impl ClockBoundServer {
             max_clock_error,
         );
 
-        if let Some(client_path) = client.as_pathname() {
-            match self.socket.send_to(&mut response, client_path) {
-                Ok(..) => return Ok(()),
-                Err(e) => warn!("Failed to send response to client. Error: {:?}", e),
-            }
-        } else {
-            // Getting the pathname may fail if an unbound socket tries to communicate with the
-            // daemon. In that case, we will drop the packet instead of attempting to reply.
-            warn!("Failed to get client pathname. Dropping packet.");
+        match self.socket.send_to_unix_addr(&mut response, &client) {
```

Comment:
It's a little messy to be returning from one branch here and not the other, when you're just returning on the next line anyway. How about just `if let  Err(e) = ... { warn!(...) }`?